### DISCUSSION TITLE
Additional I/O for nav console

### DISCUSSION THREAD

**Discussion by Cogs16** [FakeFishGames/Barotrauma#5654] [Ideas and suggestions]
It would be an immense improvment if the navigation console could output the actual speed, and depth of the submarine, which could enable depth alarms and other indicators for the rest of the crew. Additionally, inputs for controlling active/passive sonar along with autopilot would also be nice. Dividing this into an advanced terminal vs the current one may also be ok. One other thing to perhaps output is distance to destination. 

Lastly a sonar ranger that outputs the range to terrain or whatever in front of it as an additional sensor would also enable a lot of possibilities.

Great game

**Comment by James-E-A** [comment]
> a sonar ranger that outputs the range to terrain or whatever in front of it as an additional sensor

I believe you can implement this in a clunky way with a small array of motion detectors configured to only detect walls, with each detector set for a further range

**Reply by Cogs16** [reply]
We do have that, but that is very clunky and requires alot of detectors. This item would output range, bearing, and maybe position of the sensor and the type of object. This would make autonmous systems such as drones and turrets much more feasible.

**Reply by mLuby** [reply]
Suggest adding the following to the [Motion Detector](https://barotraumagame.com/wiki/Detectors#Motion_Detector):

1. Sub Editor setting for mode: **Rectangle** (default) or **Raycast**. When set to Raycast, the max x field determines max range.
2. `degrees_in` pin: controls which direction the ray goes out from the device.
3. `distance_out` pin.
4. `type_out` pin: returns the entity type id that it hits: wall, creature, sub shell, etc. (Maybe more specifically individual creatures.) Might want to use a number instead of string to avoid translation complexities.

Alternatively, give us a [Waldo/robotic manipulator utility turret](https://github.com/Regalis11/Barotrauma/discussions/7011) so we can rotate existing narrow motion sensors.

**Comment by gerakis** [comment]
I also want a pin to toggle sonar on and off, and another pin switch to auto or manual. Sometimes im away from the controls and I want to stop the ship, it would be great if I could link a wifi voice chat and just type "stop" to put the ship into standby.

I was thinking on the sonar pin 0 turns it off and 1 turns it on. The autopilot pin would have 0 is manual control, 1 is autopilot maintain pos, 2 is autopilot starting station, 3 is autopilot end station.

I know I can cut the engines and set the ballast to neutral with other components, but that's not the same as a proper maintain position. It doesn't compensate for currents and water on the ship and so forth.

**Comment by Jamessy22** [comment]
Also, customizing console signal names ingame (rather than just the editor) would be lovely - it's pretty silly having a button called "Signal #2".
